src/ValueFormatters/QuantityFormatter.php

```diff
@@ -168,4 +181,23 @@ private function formatMargin( DecimalValue $margin, $roundingExponent ) {
 		return null;
 	}
 
+	/**
+	 * @param string $number
+	 * @param string $unit
+	 *
+	 * @return string Text
+	 */
+	protected function applyUnit( $number, $unit ) {
+		if ( $this->vocabularyUriFormatter !== null ) {
+			$unit = $this->vocabularyUriFormatter->format( $unit );
+
+			if ( $unit === null ) {
+				return $number;
+			}
+		}
+
+		// TODO: localizable pattern for placement (before/after, separator)
+		return $number . ' ' . $unit;
```

Comment:
Since this depends on the locale, it seems likely that we will want to inject the logic for combining unit and number.